src/webgpu/util/conversion.ts

```diff
@@ -666,3 +686,12 @@ export function vec4(x: Scalar, y: Scalar, z: Scalar, w: Scalar) {
 
 /** Value is a Scalar or Vector value. */
 export type Value = Scalar | Vector;
+
+/** @returns if the Value is a float scalar type */
+export function isFloatValue(v: Value): boolean {
+  if (v instanceof Scalar) {
+    const s = v as Scalar;
+    return s.type.kind === s.type.kind || s.type.kind === 'f32' || s.type.kind === 'f16';
```

Comment:
Should this first one be `f64` instead of `s.type.kind`?

Reply:
Yup, straight up typo